fpptatweaks.php

```diff
@@ -489,26 +489,73 @@ function fpptatweaks_civicrm_postProcess($formName, $form) {
 function fpptatweaks_civicrm_searchColumns( $objectName, &$headers,  &$rows, &$selector ) {
   // Check if it is search contribution
   // !empty($rows) will prevent sql error if contact doesn't have contribution
-  if ($objectName == 'contribution' && !empty($rows)) {
+  if (
+    ($objectName == 'contribution'
+    || $objectName == 'event')
+    && !empty($rows)
+  ) {
+    // Since we don't want it to apply in event search, we will just return
+    if (CRM_Utils_System::getUrlPath() == 'civicrm/event/search'
+      || CRM_Utils_System::getUrlPath() == 'civicrm/contribute/search'
+    ) {
+      return;
+    }
+
     // Insert additional column header in the tab
     $insertedHeader = [
       'name' => E::ts('Inv #'),
       'field_name' => 'invoice_number',
       'direction' => CRM_Utils_Sort::DONTCARE,
       'weight' => 45,
     ];
-    $headers[] = $insertedHeader;
 
-    $contributionIds = CRM_Utils_Array::collect('contribution_id', $rows);
-    $keyPerContributionId = array_flip($contributionIds);
+    if ($objectName == 'event') {
+      // Column position of the inserted header, this is also useful
+      // ..when injecting the js for the value
+      $insertHeaderPosition = 5;
+      array_splice( $headers, $insertHeaderPosition, 0, [$insertedHeader] );
+      // Add weight on the header for the final position of the columns
+      foreach ($headers as $key => $head) {
+        $headers[$key]['weight'] = $key;
+      }
+
+      // Get participantIds array of column data
+      $participantIds = CRM_Utils_Array::collect('participant_id', $rows);
+      $contributionIds = [];
+      // Loop participantIds array to get contribution ID
+      foreach ($participantIds as $key => $participantId) {
+        $paymentDetails = civicrm_api3('ParticipantPayment', 'get', [
+          'sequential' => 1,
+          'participant_id' => $participantId,
+        ]);
+
+        // Save contributionIds as array
+        $contributionIds[$key] = $paymentDetails['values'][0]['contribution_id'];
```

Comment:
pleaes use a more descriptive variable name for $contributionIds, e.g. `$contributionIdsPerRow`, because the keys in this array (which are the keys of the $rows array) are meaningful and are used later in the code.